ftl/python/layer_builder.py

```diff
@@ -302,7 +302,26 @@ def __init__(self,
         self._cache = cache
 
     def GetCacheKeyRaw(self):
-        return '%s %s' % (self._python_cmd, self._venv_cmd)
+        python_version = self._python_version()
+        return '%s %s' % (python_version, self._venv_cmd)
+
+    def _python_version(self):
+        with ftl_util.Timing('check python version'):
+            py_version_cmd_args = [self._python_cmd, 'version']
+            proc_pipe = subprocess.Popen(
+                py_version_cmd_args,
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            stdout, stderr = proc_pipe.communicate()
+            logging.info("`python version` stdout:\n%s" % stdout)
+            if stderr:
```

Comment:
annoyingly, this won't work, because up until Python 3.4 the `version` info gets written to stderr: https://bugs.python.org/issue18338. you probably just want to check the status code here, and if it's 0 then use either stderr or stdout depending on which one isn't the empty string.